openedx/features/pakx/lms/pakx_admin_app/tasks.py

```diff
@@ -0,0 +1,73 @@
+"""Celery tasks to enroll users in courses and send registration email"""
+
+from logging import getLogger
+
+from celery import task
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.db import DatabaseError, transaction
+from edx_ace import Recipient, ace
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.ace_common.template_context import get_base_template_context
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from student.models import CourseEnrollment
+
+from .message_types import EnrolmentNotification
+from .utils import get_accessible_users
+
+log = getLogger(__name__)
+
+
+def get_enrolment_email_message_context(user, courses):
+    """
+    return context for course enrolment notification email body
+    """
+    site = Site.objects.get_current()
+    message_context = {
+        'site_name': site.domain
+    }
+    message_context.update(get_base_template_context(site))
+    message_context.update({
+        'platform_name': configuration_helpers.get_value('PLATFORM_NAME', settings.PLATFORM_NAME),
+        'username': user.username,
+        'courses': courses,
+    })
+    return message_context
+
+
+@task(name='send course enrolment email')
+def send_course_enrolment_email(users, course_keys):
+    """
+    send a course enrolment notification via email
+    """
+    courses = ', '.join(course_keys)
+    for user in users:
+        message = EnrolmentNotification().personalize(
+            recipient=Recipient(user.username, user.email),
```

Comment:
since this will be a celery task, it will only receive JSON serializable params, not objects.